frontend/src/components/shared/FeedbackDialog.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { FormEvent, useState } from 'react';
+import Dialog, { DialogProps } from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { withStyles } from '@material-ui/core/styles';
+import Radio from '@material-ui/core/Radio';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Rating from '@material-ui/lab/Rating';
+import { Locale } from 'scenes/types';
+import { Connection } from 'twilio-client';
+import { PrimaryButton } from './RoundedButton';
+
+interface FeedbackDialogProps extends DialogProps {
+  connection: Connection;
+  locale: Locale;
+}
+
+// Reference for Problem Keys: https://www.twilio.com/docs/voice/sdks/javascript/overview-1x-deprecated/connection
+const EN_STRINGS = {
+  FEEDBACK_DIALOG_TITLE: 'How was your call?',
+  FEEDBACK_SUBMIT_LABEL: 'Submit',
+  FEEDBACK_TITLES: {
+    GOOD: 'GOOD',
+    BAD: 'BAD',
+  },
+  PROBLEM_TITLES: {
+    'Call got disconnected': Connection.FeedbackIssue.DroppedCall,
+    'Lag in audio': Connection.FeedbackIssue.AudioLatency,
+    "Can't hear one another": Connection.FeedbackIssue.OneWayAudio,
+    'Sound breaking up': Connection.FeedbackIssue.ChoppyAudio,
+    'Poor audio quality': Connection.FeedbackIssue.NoisyCall,
+    'Audible echos during call': Connection.FeedbackIssue.Echo,
+  },
+};
+
+const STRINGS = {
+  en: EN_STRINGS,
+  bn: {
+    ...EN_STRINGS,
+  },
+};
+
+// TODO move this to its own module
+const withDialogButtonStyles = withStyles(() => ({
+  root: {
+    padding: '1em 2em',
+    flex: '1 0',
+    margin: '0 0.5rem',
+  },
+}));
+
+const DialogPrimaryButton = withDialogButtonStyles(PrimaryButton);
+
+export default function FeedbackDialog({
+  connection,
+  locale,
+  open,
+  onClose,
+}: FeedbackDialogProps) {
+  const [feedback, setFeedback] = useState(-1);
+  const [hover, setHover] = useState(-1);
```

Comment:
We set `hover`, but where is `hover` read?

Reply:
Actually, it is not read because we didn't implement the corresponding labelling to the score (e.g. 1 is poor, 2 is bad, etc). But the sample doc has it. Should I keep it or remove it?

Reply:
If it is not read, let's remove it.